Add security notice to joomla/crypt

paragonie-scott opened the pull request:
Depends on #8406 for its advice to hold any validity.

paragonie-scott committed:
Add security notice to joomla/crypt

paragonie-scott commented:
Not sure why this failed (most likely a runtime error with Travis CI).

paragonie-scott committed:
Update README.md

tarcieri commented on libraries/joomla/crypt/README.md:
```diff
@@ -0,0 +1,48 @@
+* `JcryptCipherMcrypt`
+* `JcryptCipherRijndael256`
+
+All of these ciphers are vulnerable to something called a [chosen-ciphertext attack](https://en.wikipedia.org/wiki/Chosen-ciphertext_attack). The only provable way to prevent chosen-ciphertext attacks is to [use uthenticated encryption](https://paragonie.com/blog/2015/05/using-encryption-and-authentication-correctly), preferrably in an [Encrypt-then-MAC construction](http://www.thoughtcrime.org/b ...
```
`uthenticated`? :stuck_out_tongue_winking_eye:

paragonie-scott replied:
Whoops. :)

paragonie-scott committed:
Actually update the example code

(The rest of the stuff needed to be updated *anyway*, but I totally forgot to do my stated reason for updating this in the first place.)

paragonie-scott committed:
Fix typo.

roland-d commented on libraries/joomla/crypt/README.md:
````diff
@@ -0,0 +1,48 @@
+```php
+$cipher = new JCryptCipherCrypto();
+$key = $cipher->generateKey(); // Store this for long-term use
+$ciphertext = $cipher->encrypt($message, $key);
````
@paragonie-scott The $message is not defined. Do you want to add that to the readme?

paragonie-scott replied:
Sure.

paragonie-scott committed:
Add example text.

paragonie-scott commented:
Does anyone object to adding this README?

Bakual commented:
No objections from me. Setting RTC from review.

Bakual commented:
@joomla-cms-bot Behave! :smile:

Kubik-Rubik commented:
Thank you @paragonie-scott! Merged.
